src/components/Action/Action.vue

```diff
@@ -0,0 +1,127 @@
+<!--
+  - @copyright Copyright (c) 2018 John Molakvoæ <skjnldsv@protonmail.com>
+  -
+  - @author John Molakvoæ <skjnldsv@protonmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<!-- if only one action, check if we need to bind to click or not -->
+	<action :href="isSingleAction && actions[0].href ? actions[0].href : '#'"
+		:class="[isSingleAction ? `${actions[0].icon} action-item--single` : 'action-item--multiple']"
+		v-bind="mainActionElement()" class="action-item"
+		v-on="isSingleAction && actions[0].action ? { click: actions[0].action } : {}">
+
+		<!-- If more than one action, create a popovermenu -->
+		<template v-if="!isSingleAction">
+			<span v-click-outside="closeMenu" tabindex="1" class="action-item__menutoggle icon-more"
+				@click="toggleMenu" />
+			<span :class="{ 'open': opened }" class="action-item__menu popovermenu">
+				<popover-menu :menu="actions" />
+			</span>
+		</template>
+
+	</action>
+</template>
+
+<script>
+import ClickOutside from 'vue-click-outside'
+import PopoverMenu from '../PopoverMenu'
+
+export default {
+	name: 'Action',
+	components: {
+		PopoverMenu
+	},
+	directives: {
+		ClickOutside
+	},
+	props: {
+		actions: {
+			type: Array,
+			required: true,
```

Comment:
does `required: true` *and* a default make any sense? I guess the default will never be used.

Reply:
I usually use the default as an example, because let's face it, no one will ever use the default yes :stuck_out_tongue_closed_eyes:

Reply:
I see. Let's move that into examples and/or documentation then :wink:

Reply:
Later then, on the jsDoc one?

Reply:
Yes!